test/util/test_case.cpp

```diff
@@ -21,6 +21,45 @@
 #include "ngraph/assertion.hpp"
 #include "test_tools.hpp"
 
+namespace
+{
+    template <typename T>
+    typename std::enable_if<std::is_floating_point<T>::value, ::testing::AssertionResult>::type
+        compare_values(const std::shared_ptr<ngraph::op::Constant>& expected_results,
+                       const std::shared_ptr<ngraph::runtime::Tensor>& results)
+    {
+        const auto expected = expected_results->get_vector<T>();
+        const auto result = read_vector<T>(results);
+        return ngraph::test::all_close_f(expected, result);
+    }
+
+    template <typename T>
+    typename std::enable_if<std::is_integral<T>::value, ::testing::AssertionResult>::type
+        compare_values(const std::shared_ptr<ngraph::op::Constant>& expected_results,
+                       const std::shared_ptr<ngraph::runtime::Tensor>& results)
+    {
+        const auto expected = expected_results->get_vector<T>();
```

Comment:
At this stage, we may need to have access to `NgraphTestCase` member variables. Let's move this logic into the class.